src/storage.ts

```diff
@@ -1,13 +1,50 @@
-type IStorage = {
-    count: number;
-};
-
-const defaultStorage: IStorage = {
-    count: 0,
-};
-
-export const storage = {
-    get: (): Promise<IStorage> =>
-        chrome.storage.sync.get(defaultStorage) as Promise<IStorage>,
-    set: (value: IStorage): Promise<void> => chrome.storage.sync.set(value),
-};
+import { writable, type Writable } from 'svelte/store';
+
+/**
+ * Creates a persistent Svelte store backed by Chrome's sync storage.
+ * @template T The type of the store's value
+ * @param key The key to use in Chrome's storage
+ * @param initialValue The initial value of the store
+ * @returns A writable Svelte store
+ */
+export function persistentStore<T>(key: string, initialValue: T): Writable<T> {
+    const store = writable(initialValue);
+    let isUpdatingFromChrome = false;
+    let isUpdatingFromStore = false;
+
+    function watchStore() {
+        store.subscribe((value) => {
+            // Prevent circular updates
+            if (isUpdatingFromChrome) return;
+            isUpdatingFromStore = true;
+            chrome.storage.sync.set({ [key]: value }).then(() => {
+                isUpdatingFromStore = false;
+            });
+        });
+    }
+
+    function watchChrome() {
+        chrome.storage.onChanged.addListener((changes, namespace) => {
+            // Prevent circular updates
+            if (isUpdatingFromStore || namespace !== 'sync' || !(Object.hasOwn(changes, key))) return;
```

Comment:
It seems that you can use `chrome.storage.sync.get` and get rid of the check `namespace !== 'sync'` 🤔

Reply:
Yep, that's a great idea.